pandas/core/dtypes/cast.py

```diff
@@ -489,11 +484,17 @@ def _maybe_cast_to_extension_array(
     -------
     ExtensionArray or obj
     """
-    from pandas.core.arrays.string_ import BaseStringArray
 
-    # Everything can be converted to StringArrays, but we may not want to convert
-    if issubclass(cls, BaseStringArray) and lib.infer_dtype(obj) != "string":
-        return obj
+    if hasattr(cls, "_from_scalars"):
+        # TODO: get this everywhere!
```

Comment:
You can probably add a simple `_from_scalars` in the base class that just calls `_from_sequence`. We will need some fallback like that for external EAs anyway (unless we keep this hasattr check, but since we fallback to `_from_sequence` below anyway, can better do this in `_from_scalars` I think)

Reply:
Agreed this makes sense.

ATM _from_sequence can raise anything, so we catch anything, which isn't an ideal pattern.  Thoughts on documenting _from_scalars as only raising ValueError/TypeError and eventually enforcing that?

Reply:
> Thoughts on documenting _from_scalars as only raising ValueError/TypeError and eventually enforcing that?

Yes, I think that's certainly fine